metrics/density/docker_memory_usage.sh

```diff
@@ -120,6 +123,59 @@ function get_pss_memory(){
 	echo "$avg"
 }
 
+# Wait for KSM to settle down, or timeout waiting
+# The basic algorithm is to look at the pages_shared value
+# at the end of every 'full scan', and if the value
+# has changed very little, then we are done (because we presume
+# a full scan has managed to do few new merges)
+#
+# arg1 - timeout in seconds
+wait_ksm_settle(){
+	local t pcnt
+	local oldscan=-1 newscan
+	local oldpages=-1 newpages
+
+	oldscan=$(cat /sys/kernel/mm/ksm/full_scans)
+
+	# Go around the loop until either we see a small % change
+	# between two full_scans, or we timeout
+	for ((t=0; t<$1; t++)); do
+
+		newscan=$(cat /sys/kernel/mm/ksm/full_scans)
+		if (( newscan != oldscan )); then
+			echo ""
```

Comment:
this could be in order to print a newline or maybe could be a typo?

Reply:
correct the first guess - it is to put out a newline to terminate the 'progress dots' generated by the line `echo -n "."` further down :-) I could comment it or make it more obvious with an `echo -n -e "\n"` if you want?
Or maybe better, I can change the line below to:
`echo -e "\nnew full_scan ($oldscan to $newscan)"`

let me do that one - that's probably the most succinct.

Reply:
updated and pushed